tests/test_autodiff.py

```diff
@@ -0,0 +1,103 @@
+"""Test the autodiff functionnality."""
+
+import numpy as np
+from mrinufft.operators.interfaces.nudft_numpy import get_fourier_matrix
+import pytest
+from pytest_cases import parametrize_with_cases, parametrize, fixture
+from case_trajectories import CasesTrajectories
+from mrinufft.operators import get_operator
+
+
+from helpers import (
+    kspace_from_op,
+    image_from_op,
+    to_interface,
+)
+
+
+TORCH_AVAILABLE = True
+try:
+    import torch
+except ImportError:
+    TORCH_AVAILABLE = False
+
+
+@fixture(scope="module")
+@parametrize(backend=["cufinufft", "finufft"])
+@parametrize(autograd=["data"])
+@parametrize_with_cases(
+    "kspace_loc, shape",
+    cases=[
+        CasesTrajectories.case_grid2D,
+        CasesTrajectories.case_nyquist_radial2D,
+    ],  # 2D cases only for reduced memory footprint.
+)
+def operator(kspace_loc, shape, backend, autograd):
+    """Create NUFFT operator with autodiff capabilities."""
+    kspace_loc = kspace_loc.astype(np.float32)
+
+    nufft = get_operator(backend_name=backend, autograd=autograd)(
+        samples=kspace_loc,
+        shape=shape,
+        smaps=None,
+        squeeze_dims=False,  # Squeezing breaks dimensions ! 
```

Comment:
@paquiteau I think we need a check for this. We need squeeze_dims to be false for now if we want autograd to work good.

Reply:
Added check for this.